kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/DekManager.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.dek;
+
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import io.kroxylicious.kms.service.Kms;
+import io.kroxylicious.kms.service.KmsService;
+import io.kroxylicious.kms.service.Serde;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A DekManager encapsulates a {@link Kms}, providing access to the ability to
+ * encrypt and decrypt using key material from the KMS without exposing
+ * that key material outside this package.
+ * @param <K> The type of KEK id
+ * @param <E> The type of encrypted DEK
+ */
+@ThreadSafe
+public class DekManager<K, E> {
+
+    private final Kms<K, E> kms;
+    private final int maxEncryptionsPerDek;
```

Comment:
use long so we can represent 2^32 (without fiddling with signed/unsigned int)

Reply:
I guess that's not a bad idea. I did also wonder if the `CipherSpec` should have a property for what the upper bound can be for a particular cipher, and allow the user to configure something lower than that if they wanted.